quantus/metrics/base.py

```diff
@@ -98,8 +129,28 @@ def __init__(
 
         self.a_axes: Sequence[int] = None
 
-        self.last_results: Any = []
-        self.all_results: Any = []
+        self.evaluation_scores: Any = []
+        self.all_evaluation_scores: Any = []
+
+    @property
+    @abstractmethod
+    def name(self):
+        raise NotImplementedError
+
+    @property
+    @abstractmethod
+    def evaluation_category(self):
+        raise NotImplementedError
+
+    @property
```

Comment:
This one can safely be implemented in the base class, 
```python
@property
def model_applicability(self) -> Set[ModelType]:
     return {ModelType.Torch, ModelType.TensorFlow}
```
Since all metrics support both, afaik.

Reply:
Sure, can be done.